### PR TITLE
Feature/upgrade

### PR DESCRIPTION
**Pull request by hellensoloviy**
Add many changes to library;

### CODE REVIEW

File: Example/Podfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
 use_frameworks!
 target 'Example' do
-  pod 'PermissionsService', :path => '../'
+  pod 'PermissionsService', :path => '../', :branch => 'feature/upgrade'
```
Comment (OverSwift) [inline Example/Podfile]: remove specified branch

File: PermissionsService/Classes/Extensions.swift [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                case 3: return .authorized
+            default: return .unknown
+            
+        }
+        
+    }
+    
+}
+
+extension Optional {
```
Comment (OverSwift) [inline PermissionsService/Classes/Extensions.swift]: Remove this extension at all and leave a system app crash which is more informative for user.

File: PermissionsService/Classes/Notifications.swift [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            
+        }
+    
+    }
+    
+    /**
+     
+     */
+    
+    public func status() -> PermissionStatus {
```
Comment (OverSwift) [inline PermissionsService/Classes/Notifications.swift:38]: why status is notDeterminated all the time

File: PermissionsService/Classes/PermissionConfiguration.swift [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    var messages: ServiceMessages { get }
+    
+}
+
+extension PermissionConfiguration {
+    public var messages: ServiceMessages { return DefaultMessages() }
+    
+}
+
+//extension PermissionConfiguration where Self: LocationConfiguration {
```
Comment (OverSwift) [inline PermissionsService/Classes/PermissionConfiguration.swift]: Is this code still needed?

**Comment by hellensoloviy** [comment]
Check up last changes, please

**Comment by hellensoloviy** [comment]
Check up last changes, please 
Now it should be ok :)

**Pull request merged**